src/dbus_mpris.rs

```diff
@@ -438,42 +439,54 @@ async fn create_dbus_server(
         let mut seeked_position = None;
 
         // Update playback state from event
-        let (track_id, playback_status) = match event {
+        let (track_id, playback_status, player_volume) = match event {
+            PlayerEvent::VolumeSet { volume } => {
+                (last_track_id, last_playback_status, Some(volume))
+            }
             PlayerEvent::Playing { track_id, position_ms, .. } => {
                 seeked_position = Some(position_ms);
-                (Some(track_id), Some(PlaybackStatus::Playing))
+                (Some(track_id), Some(PlaybackStatus::Playing), last_volume)
             }
             PlayerEvent::Stopped { .. } => {
-                (last_track_id, Some(PlaybackStatus::Stopped))
+                (last_track_id, Some(PlaybackStatus::Stopped), last_volume)
             }
             PlayerEvent::Paused { .. } => {
-                (last_track_id, Some(PlaybackStatus::Paused))
+                (last_track_id, Some(PlaybackStatus::Paused), last_volume)
             }
-            _ => (last_track_id, last_playback_status),
+            _ => (last_track_id, last_playback_status, last_volume),
         };
 
         // if playback_status or track_id have changed, emit a PropertiesChanged signal
-        if last_playback_status != playback_status || last_track_id != track_id {
+        if last_playback_status != playback_status || last_track_id != track_id || last_volume != player_volume {
             let mut changed_properties: HashMap<String, Variant<Box<dyn RefArg>>> = HashMap::new();
-            if let Some(track_id) = track_id {
-                let sp = create_spotify_api(&api_token);
-                let track = sp.track(&track_id.to_base62());
-                if let Ok(track) = track {
-                    let mut m: HashMap<String, Variant<Box<dyn RefArg>>> = HashMap::new();
-                    insert_metadata(&mut m, track);
-
-                    changed_properties.insert("Metadata".to_owned(), Variant(Box::new(m)));
-                } else {
-                    info!("Couldn't fetch metadata from spotify: {:?}", track);
+
+            if last_volume != player_volume {
+                if let Some(player_volume) = player_volume {
+                    changed_properties.insert("Volume".to_owned(), Variant(Box::new(player_volume.to_owned())));
```

Comment:
The `player_volume` here is a `u8` with values that vary from 0 (silent) to 65535 (max volume). (At least my previous observations say so) This is inconsistent with the volume that is returned from the `.Volume` property (which [appears to be](https://github.com/Spotifyd/spotifyd/blob/603bcc4d802d2b8f45ac517c0513eb4096252d7b/src/dbus_mpris.rs#L298-L306) a float from 0.0 to 100.0).

If my assumption that `librespot` just maps the volume to the full available range in a `u8`, is correct (probably should check the librespot code, if this is the case), converting this to a percent value shouldn't be too difficult.

*Note:* In fact, both "scales" don't really comply with the actual MPRIS spec, which [recommends a value from 0.0 to 1.0](https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Simple-Type:Volume). Don't know however, wether this is something that should be fixed here.

Reply:
Yep. That's true, I believe a double from 0.0 to 1.0 is the way to go